server/sonar-server/src/main/java/org/sonar/server/measure/ws/MetricDtoWithBestValue.java

```diff
@@ -57,25 +64,37 @@ MeasureDto getBestValue() {
     return bestValue;
   }
 
+  static java.util.function.Function<MetricDto, MeasureDto> buildBestMeasure(ComponentDto component, Collection<WsMeasures.Period> periods) {
+    return metric -> {
+      MeasureDto measure = new MeasureDto()
+        .setMetricId(metric.getId())
+        .setComponentUuid(component.uuid());
+      boolean isNewTypeMetric = metric.getKey().toLowerCase(Locale.ENGLISH).startsWith(LOWER_CASE_NEW_METRIC_PREFIX);
+      if (isNewTypeMetric) {
+        periods.stream()
```

Comment:
the best value is set only on the periods where we have done an analysis. For example, for the first analysis of a project, there's no period so we shouldn't set a best value on a period